src/main/scala/analysis/MemoryRegionAnalysis.scala

```diff
@@ -1,29 +1,32 @@
 package analysis
 
-import analysis.solvers.WorklistFixpointSolverWithReachability
+import analysis.BitVectorEval.isNegative
+import analysis.solvers.SimpleWorklistFixpointSolver
 import ir.*
 import util.Logger
 
 import scala.collection.mutable
 import scala.collection.mutable.ListBuffer
 
 trait MemoryRegionAnalysis(val program: Program,
+                           val domain: Set[CFGPosition],
                            val globals: Map[BigInt, String],
                            val globalOffsets: Map[BigInt, BigInt],
                            val subroutines: Map[BigInt, String],
                            val constantProp: Map[CFGPosition, Map[Variable, FlatElement[BitVecLiteral]]],
                            val ANRResult: Map[CFGPosition, Set[Variable]],
                            val RNAResult: Map[CFGPosition, Set[Variable]],
-                           val regionAccesses: Map[CFGPosition, Map[RegisterVariableWrapper, FlatElement[Expr]]],
-                           reachingDefs: Map[CFGPosition, (Map[Variable, Set[Assign]], Map[Variable, Set[Assign]])]) {
+                           val reachingDefs: Map[CFGPosition, (Map[Variable, Set[Assign]], Map[Variable, Set[Assign]])],
+                           val graResult: Map[CFGPosition, Set[DataRegion]]) {
 
-  var mallocCount: Int = 0
+  var mallocCount: BigInt = 0
   private var stackCount: Int = 0
-  val stackMap: mutable.Map[Procedure, mutable.Map[Expr, StackRegion]] = mutable.Map()
+  val stackMap: mutable.Map[Procedure, mutable.Map[BigInt, StackRegion]] = mutable.Map()
 
-  private def nextMallocCount() = {
-    mallocCount += 1
-    s"malloc_$mallocCount"
+  private def nextMallocCount(size: BigInt) = {
+    val start = mallocCount
+    mallocCount += (size.toDouble/8).ceil.toInt + 1
+    (s"malloc_$mallocCount", start)
```

Comment:
I don't understand what this is trying to do. Why does the size have anything to do with giving each malloc a unique identifier?